### PR TITLE
fix(curriculum): Fix typography in "Set a Domain and a Range on a Scale"

### PR DESCRIPTION
**Pull request by TyMick**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Changed a spaced hyphen to an [em dash](https://english.stackexchange.com/questions/2116/when-should-i-use-an-em-dash-an-en-dash-and-a-hyphen) (—). A spaced en dash ( – ) would also be appropriate here, though it's apparently [more common in British English](https://en.wikipedia.org/wiki/Dash#En_dash_versus_em_dash), while em dash usage is more common in American English.

### CODE REVIEW

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/set-a-domain-and-a-range-on-a-scale.english.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ forumTopicId: 301491
 
 ## Description
 <section id='description'>
-By default, scales use the identity relationship - the input value maps to the output value. But scales can be much more flexible and interesting.
+By default, scales use the identity relationship&mdash;the input value maps to the output value. But scales can be much more flexible and interesting.
```
Comment (Sky020) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/set-a-domain-and-a-range-on-a-scale.english.md:11]:
```suggestion
By default, scales use the identity relationship &mdash; the input value maps to the output value. But scales can be much more flexible and interesting.
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
If these changes are pushed through, I would expect the spaced version of this. As, currently, I read it the same way one would _"good-hearted"_, but _"relationship-the"_.

**Comment by tywmick** [comment]
Hmm, come to think of it, more of the world uses British conventions for English as opposed to American conventions, and freeCodeCamp does have an international audience, so that could be an argument for using the spaced en dash instead. 👍🏼

Has it been decided previously whether to use British or American English in freeCodeCamp? Searching issues and PRs, I'm finding two precedents ([1](https://github.com/freeCodeCamp/freeCodeCamp/pull/29534#issuecomment-443425206), [2](https://github.com/freeCodeCamp/freeCodeCamp/pull/20615#issuecomment-434805139)) in favor of American English and [one](https://github.com/freeCodeCamp/freeCodeCamp/pull/20605#pullrequestreview-170350544) neutral precedent.

To summarize, our two options ([according to Wikipedia](https://en.wikipedia.org/wiki/Dash#En_dash_versus_em_dash), at least) are

- British English: "By default, scales use the identity relationship &ndash; the input value maps to the output value."
  - `relationship &ndash; the`
- American English: "By default, scales use the identity relationship&mdash;the input value maps to the output value."
  - `relationship&mdash;the`

**Comment by erictleung** [comment]
> Hmm, come to think of it, more of the world uses British conventions for English as opposed to American conventions, and freeCodeCamp does have an international audience, so that could be an argument for using the spaced en dash instead. 👍🏼
> 
> Has it been decided previously whether to use British or American English in freeCodeCamp? Searching issues and PRs, I'm finding two precedents ([1](https://github.com/freeCodeCamp/freeCodeCamp/pull/29534#issuecomment-443425206), [2](https://github.com/freeCodeCamp/freeCodeCamp/pull/20615#issuecomment-434805139)) in favor of American English and [one](https://github.com/freeCodeCamp/freeCodeCamp/pull/20605#pullrequestreview-170350544) neutral precedent.

Mmm interesting. There hasn't been a ruling on what type of English freeCodeCamp lessons is meant to be written in (https://contribute.freecodecamp.org/#/how-to-work-on-coding-challenges?id=challenge-descriptionsinstructions). It just says, "plain English".

I don't want to bikeshed this topic for too long. So even though I'm in America, I'm okay with the argument that freeCodeCamp is international and should use the more widely used spaced dash.

**Comment by RandellDawson** [comment]
Let's get an official ruling from @QuincyLarson on this, then we can update the [How to work on coding challenges](https://contribute.freecodecamp.org/#/how-to-work-on-coding-challenges) documentation . 

It does not matter I suppose, but It has always been my understanding that American English spelling and grammar was to be used through out the English content of the site.  For example, we refer to a font's "color" in all the CSS challenges instead of "colour".

**Comment by thecodingaviator** [comment]
I believe that the usage still isn't standard for the `-` in much of the world. Before I started reading a lot of technical articles and such, I didn't know what the em-dash was. I had only ever seen it as a hyphen like "good-working."

Why not just use a colon?

I do wonder though if that amount of discussion this PR is needing might nudge it into bikeshedding?

**Comment by erictleung** [comment]
> Will this work on other keyboards around the world? Is the - character standard enough?

The ` - ` should be sufficient if we were able to enforce this for its use. I wasn't aware of this variation on dashs to function similar to an em-dash.

> Why not just use a colon?

A colon might be too formal of a punctuation when replacing a dash.

> I do wonder though if that amount of discussion this PR is needing might nudge it into bikeshedding?

It might be 😄 

One proposition that side steps this whole discussion is just to avoid these markers when possible. They are nice when they are used correctly, but you can just avoid them altogether.

For this PR, we could make this change:

```diff
- By default, scales use the identity relationship - the input value maps to the output value.
+ By default, scales use the identity relationship. This means the input value maps to the output value.
```

It makes descriptions more verbose, we now we don't have to deal with this "debate" on em-dash vs colon vs dash with spaces.

Thoughts?

**Comment by thecodingaviator** [comment]
@erictleung that sounds better to me!

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/set-a-domain-and-a-range-on-a-scale.english.md (hunk 1 above)
Comment (moT01) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/set-a-domain-and-a-range-on-a-scale.english.md:11]:
```suggestion
By default, scales use the identity relationship. This means the input value maps to the output value. But scales can be much more flexible and interesting.
```

**Comment by moT01** [comment]
I have added @erictleung's suggestion and resolved a conflict.

**Review by moT01 — DISMISSED** [review]
LGTM 🎉

**Review by RandellDawson — DISMISSED** [review]
LGTM

File: curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/set-a-domain-and-a-range-on-a-scale.english.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ forumTopicId: 301491
 
 ## Description
 <section id='description'>
-By default, scales use the identity relationship - the input value maps to the output value. But scales can be much more flexible and interesting.
+By default, scales use the identity relationship. This means the input value maps to the output value. But scales can be much more flexible and interesting.
 Say a data set has values ranging from 50 to 480. This is the input information for a scale, also known as the <dfn>domain</dfn>.
```
Comment (Sky020) [inline curriculum/challenges/english/04-data-visualization/data-visualization-with-d3/set-a-domain-and-a-range-on-a-scale.english.md:12]:
```suggestion
By default, scales use the identity relationship. This means the input value maps to the output value. However, scales can be much more flexible and interesting.
Say a dataset has values ranging from 50 to 480. This is the input information for a scale, also known as the <dfn>domain</dfn>.
```

**Review by thecodingaviator — DISMISSED** [review]
LGTM

**Comment by RandellDawson** [comment]
Still looks good.

**Comment by RandellDawson** [comment]
@Sky020 You still have a review showing changes requested.  Please merge if you are good with the changes.  Thanks.

**Review by ShaunSHamilton — APPROVED** [review]
